### DISCUSSION TITLE
Does it support Socket

### DISCUSSION THREAD

**Discussion by JustGigii** [exodrifter/unity-python#28] [Q&A] [answered]
Hii I tried to import Socket module.
Version of Unity 2019.3.1, Windows
Problem encountered:
`DirectoryNotFoundException: Could not find a part of the path "\Assets\Python\IronPython-2.7.10\IronPython.Modules.xml".
System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) (at <437ba245d8404784b9fbab9b439ac908>:0)
System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize) (at <437ba245d8404784b9fbab9b439ac908>:0)
(wrapper remoting-invoke-with-check) System.IO.FileStream..ctor(string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare,int)
System.Xml.XmlDownloadManager.GetStream (System.Uri uri, System.Net.ICredentials credentials, System.Net.IWebProxy proxy, System.Net.Cache.RequestCachePolicy cachePolicy) (at <1d98d70bb7d8453b80c25aa561fdecd1>:0)
System.Xml.XmlUrlResolver.GetEntity (System.Uri absoluteUri, System.String role, System.Type ofObjectToReturn) (at <1d98d70bb7d8453b80c25aa561fdecd1>:0)
System.Xml.XmlTextReaderImpl.OpenUrlDelegate (System.Object xmlResolver) (at <1d98d70bb7d8453b80c25aa561fdecd1>:0)
System.Xml.XmlTextReaderImpl.OpenUrl () (at <1d98d70bb7d8453b80c25aa561fdecd1>:0)
System.Xml.XmlTextReaderImpl.Read () (at <1d98d70bb7d8453b80c25aa561fdecd1>:0)
System.Xml.XPath.XPathDocument.LoadFromReader (System.Xml.XmlReader reader, System.Xml.XmlSpace space) (at <1d98d70bb7d8453b80c25aa561fdecd1>:0)
System.Xml.XPath.XPathDocument..ctor (System.String uri, System.Xml.XmlSpace space) (at <1d98d70bb7d8453b80c25aa561fdecd1>:0)
System.Xml.XPath.XPathDocument..ctor (System.String uri) (at <1d98d70bb7d8453b80c25aa561fdecd1>:0)
IronPython.Runtime.Types.DocBuilder.GetXPathDocument (System.Reflection.Assembly asm) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Types.DocBuilder.GetXmlDoc (System.Reflection.MethodBase info, System.String& summary, System.String& returns, System.Collections.Generic.List`1[System.Collections.Generic.KeyValuePair`2[System.String,System.String]]& parameters) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Types.DocBuilder.GetOverloadDoc (System.Reflection.MethodBase info, System.String name, System.Int32 endParamSkip, System.Boolean includeSelf) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Types.DocBuilder.CreateAutoDoc (System.Reflection.MethodBase info, System.String name, System.Int32 endParamSkip, System.Boolean includeSelf) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Types.DocBuilder.DocOneInfo (System.Reflection.MethodBase info, System.String name, System.Boolean includeSelf) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Types.DocBuilder.DocOneInfo (System.Reflection.MethodBase info, System.String name) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Types.BuiltinFunction.get___doc__ () (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Types.BuiltinMethodDescriptor.get___doc__ () (at <264a665625104e3c9b8ba1905cf1b515>:0)
(wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,object)
Microsoft.Scripting.Interpreter.DynamicInstruction`2[T0,TRet].Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run3[T0,T1,T2,TRet] (T0 arg0, T1 arg1, T2 arg2) (at <adbce494b74f4c8787535955fc342591>:0)
IronPython.Compiler.Ast.DynamicGetMemberExpression+GetMemberInstruction.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <264a665625104e3c9b8ba1905cf1b515>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run1[T0,TRet] (T0 arg0) (at <adbce494b74f4c8787535955fc342591>:0)
IronPython.Compiler.RuntimeScriptCode.InvokeTarget (Microsoft.Scripting.Runtime.Scope scope) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.RuntimeScriptCode.Run (Microsoft.Scripting.Runtime.Scope scope) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.PythonContext.InitializeModule (System.String fileName, IronPython.Runtime.ModuleContext moduleContext, Microsoft.Scripting.ScriptCode scriptCode, IronPython.Runtime.ModuleOptions options) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.PythonContext.CompileModule (System.String fileName, System.String moduleName, Microsoft.Scripting.SourceUnit sourceCode, IronPython.Runtime.ModuleOptions options, Microsoft.Scripting.ScriptCode& scriptCode) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.PythonContext.CompileModule (System.String fileName, System.String moduleName, Microsoft.Scripting.SourceUnit sourceCode, IronPython.Runtime.ModuleOptions options) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Importer.LoadFromSourceUnit (IronPython.Runtime.CodeContext context, Microsoft.Scripting.SourceUnit sourceCode, System.String name, System.String path) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Importer.LoadModuleFromSource (IronPython.Runtime.CodeContext context, System.String name, System.String path) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Importer.LoadFromDisk (IronPython.Runtime.CodeContext context, System.String name, System.String fullName, System.String str) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Importer.ImportFromPathHook (IronPython.Runtime.CodeContext context, System.String name, System.String fullName, IronPython.Runtime.List path, System.Func`5[T1,T2,T3,T4,TResult] defaultLoader) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Importer.ImportFromPath (IronPython.Runtime.CodeContext context, System.String name, System.String fullName, IronPython.Runtime.List path) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Importer.ImportTopAbsolute (IronPython.Runtime.CodeContext context, System.String name) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Importer.ImportModule (IronPython.Runtime.CodeContext context, System.Object globals, System.String modName, System.Boolean bottom, System.Int32 level) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Modules.Builtin.__import__ (IronPython.Runtime.CodeContext context, System.String name, System.Object globals, System.Object locals, System.Object fromlist, System.Int32 level) (at <264a665625104e3c9b8ba1905cf1b515>:0)
Microsoft.Scripting.Interpreter.FuncCallInstruction`7[T0,T1,T2,T3,T4,T5,TRet].Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run7[T0,T1,T2,T3,T4,T5,T6,TRet] (T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6) (at <adbce494b74f4c8787535955fc342591>:0)
System.Dynamic.UpdateDelegates.UpdateAndExecute6[T0,T1,T2,T3,T4,T5,TRet] (System.Runtime.CompilerServices.CallSite site, T0 arg0, T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5) (at <351e49e2a5bf4fd6beabb458ce2255f3>:0)
IronPython.Runtime.Importer.ImportLightThrow (IronPython.Runtime.CodeContext context, System.String fullName, IronPython.Runtime.PythonTuple from, System.Int32 level) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Runtime.Operations.PythonOps.ImportTop (IronPython.Runtime.CodeContext context, System.String fullName, System.Int32 level) (at <264a665625104e3c9b8ba1905cf1b515>:0)
Microsoft.Scripting.Interpreter.FuncCallInstruction`4[T0,T1,T2,TRet].Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.Interpreter.Run (Microsoft.Scripting.Interpreter.InterpretedFrame frame) (at <adbce494b74f4c8787535955fc342591>:0)
Microsoft.Scripting.Interpreter.LightLambda.Run2[T0,T1,TRet] (T0 arg0, T1 arg1) (at <adbce494b74f4c8787535955fc342591>:0)
IronPython.Compiler.PythonScriptCode.RunWorker (IronPython.Runtime.CodeContext ctx) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.PythonScriptCode.Run (Microsoft.Scripting.Runtime.Scope scope) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.RuntimeScriptCode.InvokeTarget (Microsoft.Scripting.Runtime.Scope scope) (at <264a665625104e3c9b8ba1905cf1b515>:0)
IronPython.Compiler.RuntimeScriptCode.Run (Microsoft.Scripting.Runtime.Scope scope) (at <264a665625104e3c9b8ba1905cf1b515>:0)
Microsoft.Scripting.SourceUnit.Execute (Microsoft.Scripting.Runtime.Scope scope, Microsoft.Scripting.ErrorSink errorSink) (at <96bae61c64314af5927380c0b6bd16cf>:0)
Microsoft.Scripting.SourceUnit.Execute (Microsoft.Scripting.Runtime.Scope scope) (at <96bae61c64314af5927380c0b6bd16cf>:0)
Microsoft.Scripting.Hosting.ScriptSource.Execute (Microsoft.Scripting.Hosting.ScriptScope scope) (at <96bae61c64314af5927380c0b6bd16cf>:0)
Microsoft.Scripting.Hosting.ScriptEngine.ExecuteFile (System.String path, Microsoft.Scripting.Hosting.ScriptScope scope) (at <96bae61c64314af5927380c0b6bd16cf>:0)
Microsoft.Scripting.Hosting.ScriptEngine.ExecuteFile (System.String path) (at <96bae61c64314af5927380c0b6bd16cf>:0)`

**Comment by exodrifter** [comment]
Where did you place the code in your project? It looks like it can't find a file; if you just copied the example without updating the path to where you actually put the files, it won't be able to find it.

If you are still running into a problem, please share the search paths you set for the engine (you can get these by calling `engine.GetSearchPaths();` and where you placed `unity-python` in the Assets folder.

**Comment by JustGigii** [comment]
When I change the import socket to import Theard. It's work. But if I add import socket in blank script. I see the problem again

**Comment by JustGigii** [comment]
![Screenshot_2](https://user-images.githubusercontent.com/46539955/81053132-4a0c9580-8ecd-11ea-938a-7016723a54df.png)
![Screenshot_1](https://user-images.githubusercontent.com/46539955/81053134-4aa52c00-8ecd-11ea-9a60-826cb44807ae.png)

**Comment by exodrifter** [comment, marked as answer]
I tested this on Unity 2019.3.12f1 and Unity 2019.3.1f1 on Windows, and importing `socket` works in both versions. Maybe you accidentally deleted `IronPython.Modules.xml`? Anyway, give the attached project a try by running `Assets/Scenes/SampleScene.unity`: [test.zip](https://github.com/exodrifter/unity-python/files/4588938/test.zip)

If it works, then there must be something wrong in your project. Otherwise, there might be something weird about how your system is configured. If you're still unable to figure out the issue, please post a minimal project that reproduces the issue.

**Comment by JustGigii** [comment]
Thank you for the help. I was play with the files And The problem solve. in mystery way.
